### PR TITLE
Fix docs (especially for `coxph`) and UI messaging

### LINKED ISSUE #260: Problems with `survival::coxph()` and env handling
## perhaps reporting if any changes are made, even if overall object size is unchanged?

(thank you for a wonderful package and apologies if any of the below is due to my own misunderstandings!)

It seems to me that verbose = TRUE reports an unsuccessful `butcher` if the total memory of an object does not change, even though aspects of the object may successfully be butchered (I will admit to having only a very fuzzy understanding of memory).

I was confused by output when I ran into this situation and initially thought that `butcher` was not able to trim my coxph object:

``` r
library(tibble)
library(lobstr)
library(survival)
library(butcher)

example_data <-
  tibble::tibble(
    time = rpois(10000, 2) + 1,
    status = rbinom(10000, 1, .5),
    x = rpois(10000, .5),
    covar = rbinom(10000, 1, .5)
  )

coxph_formula <- function(data) {
  res <- coxph(Surv(time, status) ~ x + strata(covar), data)
  res_formula <- res$formula
}

coxph_formula_butchered <- function(data) {
  res <- coxph(Surv(time, status) ~ x + strata(covar), data)
  cleaned_res <- butcher(res, verbose = TRUE)
  res_formula <- cleaned_res$formula
}

res <- coxph_formula(example_data)
res_butchered <- coxph_formula_butchered(example_data)
#> ✖ No memory released. Do not butcher.

lobstr::obj_size(res)
#> 534.84 kB
lobstr::obj_size(res_butchered)
#> 1.49 kB
```

<sup>Created on 2023-04-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

but clearly the formula object is much reduced. I am assuming that the env (or the contents of the env) that is removed from `formula` exists in a different object, so the total size of `res_butchered` is unchanged and thus `assess_object` determines that nothing net changed. But would it be possible to report `memory_released` on the level of the implemented `axe` s as well as the overall object size?

(The message changes to the expected information about memory being released if only I had butchered the thing I was truly interested in (`res_formula <- butcher(res$formula, verbose = TRUE)`) rather than turning it into a two-step process as I did, but memory is confusing and a more informative message could have prevented a few of the rabbit holes I went down before this all made sense)

Thanks for considering this!

### PR DESCRIPTION
**Pull request by juliasilge**
Closes #260 

This PR mainly

- adds a disclaimer to the `coxph` docs about not using butcher from anywhere but the global env
- changes the UI messaging to tell folks when their butchered object is actually _larger_

Turns out we didn't have `markdown = TRUE` set yet so this PR makes that change and updates the needed docs as well.

### CODE REVIEW

File: R/ui.R [hunk 1]
```diff
@@ -15,13 +15,11 @@ memory_released <- function(og, butchered) {
   old <- lobstr::obj_size(og)
   new <- lobstr::obj_size(butchered)
   rel <- old - new
-  rel <- format(rel, big.mark = ",", scientific = FALSE)
   if (length(rel) == 1) {
-    if (rel <= 0) {
+    if (isTRUE(all.equal(old, new))) {
```
Comment (juliasilge) [inline R/ui.R:19]: Floating point math scares me. 😱 Any better options for this?

File: tests/testthat/test-ui.R [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  small <- add_butcher_class(small)
+  obj_size_diff <- lobstr::obj_size(big) - lobstr::obj_size(small)
+  obj_size_diff <- format(obj_size_diff, big.mark = ",", scientific = FALSE)
+
+  expect_snapshot({
+    assess_object(big, small)
+    assess_object(small, small)
+    assess_object(small, big)
+  },
+  transform = function(x) gsub(obj_size_diff, "<redacted>", x, fixed = TRUE)
```
Comment (juliasilge) [inline tests/testthat/test-ui.R:14]: I'm not sure if it's possible that the memory for these objects would be different in the future or on a different platform, but let's just be safe.

Comment (simonpcouch) [reply tests/testthat/test-ui.R:14]: Yup, I think that's a good idea.👍

File: R/ui.R [hunk 3]
```diff
@@ -35,13 +33,18 @@ assess_object <- function(og, butchered) {
...
-    cli::cli_alert_success("Memory released: {.val {mem}}")
-    if (!is.null(disabled)) {
-      cli::cli_alert_danger("Disabled: {.code {disabled}}")
-    }
-    if (length(class_added) == 0) {
-      class_name <- "butchered"
-      cli::cli_alert_danger("Could not add {.cls {class_name}} class")
+    abs_mem <- format(abs(mem), big.mark = ",", scientific = FALSE)
+    if (mem < 0) {
+      cli::cli_alert_danger("The butchered object is {.val {abs_mem}} larger than the original. Do not butcher.")
```
Comment (simonpcouch) [inline R/ui.R:38]: With this PR, I see:

```
> res_via_func <- coxph_formula_butchered(example_data)
✖ The butchered object is "920 B" larger than the original. Do not butcher.
```

(with the blue formatting for 920 B). The quotes caught my eye--perhaps we could use `.field` or a straight up glue substitution here?

Comment (simonpcouch) [reply R/ui.R:38]: Oh, I see that `Memory released: "533.24 kB"` uses this pattern too. Maybe this is fine!

Comment (simonpcouch) [reply R/ui.R:38]: Something about the phrasing this message feels strange to me but maybe that's the purpose—raise an eyebrow in this weird (maybe coxph-specific?) situation.

**Review by simonpcouch — APPROVED** [review]
The tests are wonderfully concise and I'm appreciating the way you've phrased these docs. Thumbs up from me. :)

File: R/ui.R (hunk 3 above)
Comment (juliasilge) [reply R/ui.R:38]: I agree with you that the BLUE plus QUOTES looks weird. What is I change them both to just `.field`? That makes them green with no quote marks.

Comment (simonpcouch) [reply R/ui.R:38]: I'm game for that!

**Pull request merged**